src/seedu/addressbook/commands/Command.java

```diff
@@ -70,4 +70,13 @@ public int getTargetIndex() {
     public void setTargetIndex(int targetIndex) {
         this.targetIndex = targetIndex;
     }
+
+/**
+ * Check if command is mutating and return true if command is mutating.
+ *
+ */
+    public boolean isMutating() {
+        throw new UnsupportedOperationException("This method should be implemented in child class");
```

Comment:
You do not have to throw any exceptions -- just make the method abstract by not including any implementation.